File: app/src/main/java/workshop1024/com/xproject/controller/adapter/TagListAdapter.java
package workshop1024.com.xproject.controller.adapter;

import android.content.Context;
import android.content.res.Resources;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import workshop1024.com.xproject.R;
import workshop1024.com.xproject.model.filter.source.FilterRepository;
import workshop1024.com.xproject.utils.UnitUtils;
import workshop1024.com.xproject.view.dialog.InputStringDialog;

public class TagListAdapter extends RecyclerView.Adapter<TagListAdapter.TagViewHolder> {
    private Context mContext;
    private Resources mResource;
    private List<String> mTagList;

    public TagListAdapter(Context context, List<String> tagList) {
        mContext = context;
        mTagList = tagList;
        mResource = context.getResources();
    }

    @Override
    public TagViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        TextView tagTextView = new TextView(parent.getContext());
        tagTextView.setBackgroundResource(R.drawable.taglist_button_selector);
        tagTextView.setTextColor(mResource.getColor(R.color.white, null));
        //TODO dp和像素的转换
        tagTextView.setTextSize(14);
        tagTextView.setPadding(UnitUtils.dpToPx(mContext, 8), UnitUtils.dpToPx(mContext, 5),
                UnitUtils.dpToPx(mContext, 8), UnitUtils.dpToPx(mContext, 5));
        return new TagViewHolder(tagTextView);
    }

    @Override
    public void onBindViewHolder(TagViewHolder holder, int position) {
        holder.mTagTextView.setText(mTagList.get(position));
    }

    @Override
    public int getItemCount() {
        return mTagList.size();
    }

    public class TagViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener,
            InputStringDialog.InputStringDialogListener {
        private TextView mTagTextView;

        public TagViewHolder(View itemView) {
            super(itemView);
            mTagTextView = (TextView) itemView;
            mTagTextView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            InputStringDialog addFilterDialog = InputStringDialog.newInstance(R.string.addfilter_dialog_title, R.string
                    .addfilter_dialog_positive);
            addFilterDialog.setPreInputString(mTagTextView.getText().toString());
            addFilterDialog.setInputStringDialogListener(this);
            //FIXME 似乎获取getSupportFragmentManager有点别扭
            addFilterDialog.show(((AppCompatActivity) mContext).getSupportFragmentManager(), "addFilterDialog");
        }

        @Override
        public void onInputStringDialogClick(DialogFragment dialog, String inputString) {
            FilterRepository filterRepository = FilterRepository.getInstance();
            filterRepository.addFilterByName(inputString);
        }
    }
}

File: home/src/main/java/workshop1024/com/xproject/home/controller/adapter/TagListAdapter.kt
package workshop1024.com.xproject.home.controller.adapter

import android.content.Context
import android.content.res.Resources
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import androidx.recyclerview.widget.RecyclerView
import workshop1024.com.xproject.base.utils.UnitUtils
import workshop1024.com.xproject.home.R
import workshop1024.com.xproject.home.model.Injection
import workshop1024.com.xproject.home.view.dialog.InputStringDialog

class TagListAdapter(private val mContext: Context, private val mTagList: List<String>) : RecyclerView.Adapter<TagListAdapter.TagViewHolder>() {

    private val mResource: Resources = mContext.resources

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TagViewHolder {
        val tagTextView = TextView(parent.context)
        tagTextView.setBackgroundResource(R.drawable.taglist_button_selector)
        tagTextView.setTextColor(mResource.getColor(R.color.newsdetail_taglist_textcolor, null))
        //TODO dp和像素的转换
        tagTextView.textSize = 14f
        tagTextView.setPadding(UnitUtils.dpToPx(mContext, 8f), UnitUtils.dpToPx(mContext, 5f),
                UnitUtils.dpToPx(mContext, 8f), UnitUtils.dpToPx(mContext, 5f))
        return TagViewHolder(tagTextView)
    }

    override fun getItemCount(): Int {
        return mTagList.size
    }

    override fun onBindViewHolder(holder: TagViewHolder, position: Int) {
        holder.mTagTextView.text = mTagList[position]
    }

    inner class TagViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener, InputStringDialog.InputStringDialogListener {
        internal val mTagTextView: TextView = itemView as TextView

        init {
            mTagTextView.setOnClickListener(this)
        }

        override fun onClick(v: View?) {
            val addFilterDialog = InputStringDialog.newInstance(R.string.addfilter_dialog_title, R.string
                    .addfilter_dialog_positive)
            addFilterDialog.mPreInputString = mTagTextView.text.toString()
            addFilterDialog.mInputStringDialogListener = this
            //FIXME 似乎获取getSupportFragmentManager有点别扭
            addFilterDialog.show((mContext as AppCompatActivity).supportFragmentManager, "addFilterDialog")
        }

        override fun onInputStringDialogClick(dialog: DialogFragment, inputString: String) {
            val filterRepository = Injection.provideFilterRepository()
            filterRepository.addFilterByName(inputString)
        }

    }
}